overlord/snapstate/snapstate_update_test.go

```diff
@@ -8228,75 +8204,51 @@ func (s *snapmgrTestSuite) TestUpdateBaseKernelSingleRebootHappy(c *C) {
+	var baseTs, kernelTs *state.TaskSet
 	for _, ts := range tss {
-		chg.AddAll(ts)
-		for _, tsk := range ts.Tasks() {
-			switch tsk.Kind() {
-			// setup-profiles should appear right before link-snap,
-			// while set-auto-aliase appears right after
-			// auto-connect
-			case "link-snap", "auto-connect", "setup-profiles", "set-auto-aliases":
-				snapsup, err := snapstate.TaskSnapSetup(tsk)
-				c.Assert(err, IsNil)
-				snapTasks[fmt.Sprintf("%s@%s", tsk.Kind(), snapsup.Type)] = tsk
-				if tsk.Kind() == "link-snap" {
-					opts := 0
-					if snapsup.Type == snap.TypeBase {
-						opts |= noConfigure
-						baseTs = ts
-					} else if snapsup.Type == snap.TypeKernel {
-						kernelTs = ts
-					}
-					verifyUpdateTasks(c, snapsup.Type, opts, 0, ts)
-				}
+		for _, t := range ts.Tasks() {
+			snapsup, err := snapstate.TaskSnapSetup(t)
+			if err != nil {
+				continue
+			}
+			if snapsup.Type == snap.TypeKernel {
+				kernelTs = ts
+			} else if snapsup.Type == snap.TypeBase {
+				baseTs = ts
 			}
 		}
+		chg.AddAll(ts)
 	}
-
-	c.Assert(snapTasks, HasLen, 8)
-	linkSnapKernel := snapTasks["link-snap@kernel"]
-	autoConnectKernel := snapTasks["auto-connect@kernel"]
-	linkSnapBase := snapTasks["link-snap@base"]
-	autoConnectBase := snapTasks["auto-connect@base"]
-	c.Assert(kernelTs, NotNil)
 	c.Assert(baseTs, NotNil)
-	c.Assert(kernelTs.MaybeEdge(snapstate.BeforeMaybeRebootEdge), Equals, snapTasks["setup-profiles@kernel"])
-	c.Assert(kernelTs.MaybeEdge(snapstate.MaybeRebootEdge), Equals, linkSnapKernel)
-	c.Assert(kernelTs.MaybeEdge(snapstate.MaybeRebootWaitEdge), Equals, autoConnectKernel)
-	c.Assert(kernelTs.MaybeEdge(snapstate.AfterMaybeRebootWaitEdge), Equals, snapTasks["set-auto-aliases@kernel"])
-
-	c.Assert(baseTs.MaybeEdge(snapstate.BeforeMaybeRebootEdge), Equals, snapTasks["setup-profiles@base"])
-	c.Assert(baseTs.MaybeEdge(snapstate.MaybeRebootEdge), Equals, linkSnapBase)
-	c.Assert(baseTs.MaybeEdge(snapstate.MaybeRebootWaitEdge), Equals, autoConnectBase)
-	c.Assert(baseTs.MaybeEdge(snapstate.AfterMaybeRebootWaitEdge), Equals, snapTasks["set-auto-aliases@base"])
+	c.Assert(kernelTs, NotNil)
 
-	c.Assert(linkSnapBase.WaitTasks(), DeepEquals, []*state.Task{
-		snapTasks["setup-profiles@base"], snapTasks["setup-profiles@kernel"],
-	})
-	c.Assert(linkSnapKernel.WaitTasks(), DeepEquals, []*state.Task{
-		snapTasks["setup-profiles@kernel"], linkSnapBase,
-	})
-	c.Assert(autoConnectKernel.WaitTasks(), DeepEquals, []*state.Task{
-		snapTasks["link-snap@kernel"], autoConnectBase,
-	})
-	c.Assert(autoConnectBase.WaitTasks(), DeepEquals, []*state.Task{
-		snapTasks["link-snap@base"], snapTasks["link-snap@kernel"],
-	})
-	c.Assert(snapTasks["set-auto-aliases@kernel"].WaitTasks(), DeepEquals, []*state.Task{
-		autoConnectKernel,
-	})
-	c.Assert(snapTasks["set-auto-aliases@base"].WaitTasks(), DeepEquals, []*state.Task{
-		autoConnectBase, autoConnectKernel,
-	})
+	// Grab the tasks we need to check dependencies between
```

Comment:
all the tests that are being changes should also have checks and have assertions about the lane configuration of the change, as we are changing it or making sure we keep things as on master when appropriate

Reply:
I added some checks in the tests changed that the expected snaps to have this additional lane actually share a lane.